frontend/src/machines/billings/billingMachine.ts

```diff
@@ -0,0 +1,517 @@
+      context: Context & {
+        billings: Billing[];
+        billingError: null;
+        formMode: "CREATE" | "EDIT";
+        members: Member[];
+        membersError: null;
+        billingDetail: BillingDetail;
+        billingDetailError: null;
+        billingForm: BillingForm & {
+          title: string;
+          total: number;
+          chargedMember: string;
+          isBillEqual: boolean;
+          members: [];
+        };
+        submitBillingError: null;
+      };
+    }
+  | {
+      value: { billingFormReady: "secondStep" };
+      context: Context & {
+        billings: Billing[];
+        billingError: null;
+        formMode: "CREATE" | "EDIT";
+        members: Member[];
+        membersError: null;
+        billingDetail: BillingDetail;
+        billingDetailError: null;
+        billingForm: BillingForm & {
+          title: string;
+          total: number;
+          chargedMember: string;
+          isBillEqual: boolean;
+          members: BillingMember[];
+        };
+        submitBillingError: null;
+      };
+    }
+  | {
+      value: "submitBilling";
+      context: Context & {
+        billings: Billing[];
+        billingError: null;
+        formMode: "CREATE" | "EDIT";
+        members: Member[];
+        membersError: null;
+        billingDetail: BillingDetail;
+        billingDetailError: null;
+        billingForm: BillingForm & {
+          title: string;
+          total: number;
+          chargedMember: string;
+          isBillEqual: boolean;
+          members: BillingMember[];
+        };
+        submitBillingError: null;
+      };
+    }
+  | {
+      value: "submitBillingOK";
+      context: Context & {
+        billings: Billing[];
+        billingError: null;
+        formMode: "CREATE" | "EDIT";
+        members: Member[];
+        membersError: null;
+        billingDetail: BillingDetail;
+        billingDetailError: null;
+        billingForm: BillingForm & {
+          title: string;
+          total: number;
+          chargedMember: string;
+          isBillEqual: boolean;
+          members: BillingMember[];
+        };
+        submitBillingError: null;
+      };
+    }
+  | {
+      value: "submitBillingError";
+      context: Context & {
+        billings: Billing[];
+        billingError: null;
+        formMode: "CREATE" | "EDIT";
+        members: Member[];
+        membersError: null;
+        billingDetail: BillingDetail;
+        billingDetailError: null;
+        billingForm: BillingForm & {
+          title: string;
+          total: number;
+          chargedMember: string;
+          isBillEqual: boolean;
+          members: BillingMember[];
+        };
+        submitBillingError: string;
+      };
+    };
+
+type MachineEvents =
+  | { type: "FETCH_BILLING" }
+  | { type: "FETCH_BILLING_SUCCES"; billingsData: Billing[] }
+  | { type: "FETCH_BILLING_ERROR"; billingsErrorMessage: string }
+  | { type: "ACTIVATE_BILLING_FORM" }
```

Comment:
ini perlu ada nambahin `formMode` kayaknya ya ?  

```
{ type: "ACTIVATE_BILLING_FORM", formMode: FormMode }
```

Reply:
I didn't see that coming ka, otherwise context formMode nya ga bakal berubah.